plugins/timesync.koplugin/main.lua

```diff
@@ -0,0 +1,57 @@
+
+local Device = require("device")
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local T = require("ffi/util").template
+local _ = require("gettext")
+
+if not (Device:isKobo() or Device:isKindle() or Device:isPocketbook()) then
+    return { disabled = true, }
+end
+
+local TimeSync = WidgetContainer:new{
+    name = "timesync",
+}
+
+local function currentTime()
+    local std_out = io.popen("date")
+    if std_out then
+        local result = std_out:read("*all")
+        std_out:close()
+        return T(_("New Time is %1"), result)
+    else
+        return _("Time synchronized")
```

Comment:
I do not think the "Time synchronized" message will show up, "date" will always output something. But considering io.popen() may return a nil, and cause the following std_out:read() to fail, so fall back to show a "time synchronized" message to tell users something has been done.